### PR TITLE
PHOENIX-6191: Creating a view which has its own new columns should also do checkAndPut checks on SYSTEM.MUTEX

### PR DESCRIPTION
**Pull request by ChinmaySKulkarni**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m  7s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 41s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 57s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   1m 39s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 42s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m 28s |  phoenix-core in 4.x has 946 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 27s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  3s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  3s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 39s |  phoenix-core: The patch generated 1 new + 2824 unchanged - 4 fixed = 2825 total (was 2828)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 38s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 210m 17s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 249m 54s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.IndexBuildTimestampIT |
|   | phoenix.end2end.ViewMetadataIT |
|   | phoenix.query.MaxConcurrentConnectionsIT |
|   | phoenix.end2end.index.GlobalMutableNonTxIndexIT |
|   | phoenix.end2end.index.IndexMetadataIT |
|   | phoenix.end2end.UpsertWithSCNIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/966 |
| JIRA Issue | PHOENIX-6191 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 30ac4d9cdae4 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 565b0ea |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/1/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/1/testReport/ |
| Max. process+thread count | 6259 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java [hunk 1]
```diff
@@ -0,0 +1,793 @@
...
+        }
+    }
+
+    /**
+     * Concurrently create a view which has its own new column and also try to
+     * add the same column to its parent.
+     * See PHOENIX-6191
+     */
+    @Test
+    public void testConcurrentViewCreationWithNewColParentColAddition()
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java:599]: This is the only new test

**Comment by yanxinyi** [comment]
The new test and changes look to me. Just want to confirm formatting changes and your change is not related to the test failures above. Otherwise, +1

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java [hunk 2]
```diff
@@ -0,0 +1,793 @@
...
+                    + generateUniqueName();
+            latch1 = new CountDownLatch(1);
+            latch2 = new CountDownLatch(1);
+            String tableDdl = "CREATE TABLE " + fullTableName +
+                    "  (k INTEGER NOT NULL PRIMARY KEY, v1 INTEGER, v2 INTEGER)"
+                    + tableDDLOptions;
+            stmt.execute(tableDdl);
+
+            ExecutorService executorService = Executors.newFixedThreadPool(
+                    1, new ThreadFactory() {
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java:209]: nit: maybe use `newSingleThreadExecutor()` instead? (for all similar places in this file)

**Review by virajjasani — APPROVED** [review]
+1, ran new test locally, left one minor nit

**Comment by virajjasani** [comment]
Also, test `ViewMetadataIT.testViewAndTableAndDropCascade()` failure seem flaky and not permanent, but still good to confirm. Agree with @yanxinyi , let me spin a new build for better visibility.
Thanks

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  11m 12s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 53s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   1m 40s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  4.x passed  |
| +0 :ok: |  spotbugs  |   2m 52s |  phoenix-core in 4.x has 946 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 42s |  phoenix-core: The patch generated 1 new + 2824 unchanged - 4 fixed = 2825 total (was 2828)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  3s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 185m 49s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 37s |  The patch does not generate ASF License warnings.  |
|  |   | 219m 30s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.DeleteIT |
|   | phoenix.end2end.index.GlobalImmutableTxIndexIT |
|   | phoenix.end2end.DropIndexedColsIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/966 |
| JIRA Issue | PHOENIX-6191 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 93a77fb01df6 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 110f5b7 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/3/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/3/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/3/testReport/ |
| Max. process+thread count | 6269 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/3/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java (hunk 1 above)
Comment (jpisaac) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java:599]: @ChinmaySKulkarni What is the expected behavior when it is a global view and only one (or some tenants) have the (offending) column?

**Review by jpisaac — COMMENTED** [review]
Looking good, Had a question?

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java (hunk 1 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewConcurrencyAndFailureIT.java:599]: If any tenant view has the column already, then a conflicting column addition to its parent global view or even the ancestor base table will fail with a `CANNOT_MUTATE_TABLE` SQLException

**Comment by ChinmaySKulkarni** [comment]
Thanks for the reviews @yanxinyi @jpisaac @virajjasani. I have run the test locally many times and it didn't fail/flap. Also addressed Viraj's comment. I'll commit it if there are no more comments. Please take a look, thanks.

**Review by jpisaac — APPROVED** [review]

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  7s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 39s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 57s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   1m 40s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 42s |  4.x passed  |
| +0 :ok: |  spotbugs  |   2m 52s |  phoenix-core in 4.x has 946 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 56s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 42s |  phoenix-core: The patch generated 34 new + 2791 unchanged - 37 fixed = 2825 total (was 2828)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 42s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  5s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 192m 49s |  phoenix-core in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  The patch does not generate ASF License warnings.  |
|  |   | 225m 53s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.end2end.ViewIT |
|   | phoenix.end2end.LocalIndexSplitMergeIT |
|   | phoenix.end2end.PhoenixDriverIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/966 |
| JIRA Issue | PHOENIX-6191 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 893236d29aef 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 110f5b7 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/4/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/4/artifact/yetus-general-check/output/patch-unit-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/4/testReport/ |
| Max. process+thread count | 6327 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/4/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  5s |  https://github.com/apache/phoenix/pull/966 does not apply to 4.x. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/phoenix/pull/966 |
| JIRA Issue | PHOENIX-6191 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-966/5/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**